### PR TITLE
Update Raspberry Pi instructions

### PR DESCRIPTION
**Pull request by richlander**
This doc is intended to explain how to use .NET Core on the Raspberry Pi for Linux and Windows, ARM32 and ARM64 (where applicable).

To make things simpler, I am going to make the doc .NET Core 3.0 only and direct people to where to get links for 2.x. The product is so much easier and more pleasant to use on the Pi with 3.0 so we should go with that.

### CODE REVIEW

File: samples/RaspberryPiInstructions.md [hunk 1]
```diff
@@ -1,70 +1,104 @@
...
+It is easy to run an application on the Pi. The following commands are for Linux, but switching the casing will enable the same commands to work on Windows.
 
-### Linux
+Console App:
 
-* Install [Linux](https://www.raspberrypi.org/downloads/) on your Pi.
+1. `dotnet new console -o app` // Generate app from template
+2. `cd app`
+3. `dotnet run` // Build and run app
+4. `./bin/Debug/netcoreapp3.0/app` // Alternatively, run app directly
```
Comment (leecow) [inline samples/RaspberryPiInstructions.md]: Shouldn't this be `dotnet /bin/Debug/netcoreapp3.0/app.dll? When I try to run directly I get 

```
leecow at Brandywine in ~/test
$ ./bin/Debug/netcoreapp2.2/test
-bash: ./bin/Debug/netcoreapp2.2/test: No such file or directory
```

File: samples/RaspberryPiInstructions.md [hunk 2]
```diff
@@ -1,70 +1,104 @@
...
 
-Note: All models of generation 1 and Pi Zero are not supported because the .NET Core JIT depends on armv7 instructions not available on those versions.
+We recommend that you use .NET Core 3 for Raspberry Pi development. There are significant improvements in .NET Core 3 that make the experience much better.
 
-## Creating an app:
+Note: .NET Core supports Raspberry Pi 2 and Pi 3. We recommend the Pi 3.
 
-* [Install .NET Core SDK](https://www.microsoft.com/net/core) into a supported developer configuration.
-(Raspberry Pi itself is supported only as deployment target but there is an unsupported version of the SDK available as well.)
+Note: You can run [32-bit Linux](https://www.raspberrypi.org/downloads/raspbian/), [64-bit Linux](https://gist.github.com/richlander/467813274cea8abc624553ee72b28213#how-to-install-arm64-builds) and [32-bit Windows](https://docs.microsoft.com/en-us/windows/iot-core/downloads) on the Pi. Any of those options will work with .NET Core. On Linux, it may be helpful to [Starting ssh automatically at boot time](https://raspberrypi.stackexchange.com/questions/1747/starting-ssh-automatically-at-boot-time).
```
Comment (joperezr) [inline samples/RaspberryPiInstructions.md]: > [Starting ssh automatically at boot time](https://raspberrypi.stackexchange.com/questions/1747/starting-ssh-automatically-at-boot-time).

NIT: Should that be instead:
`[start ssh automatically at boot time](https://raspberrypi.stackexchange.com/questions/1747/starting-ssh-automatically-at-boot-time).`

File: samples/RaspberryPiInstructions.md [hunk 3]
```diff
@@ -1,70 +1,129 @@
...
+  * 2.2 ARM32: `curl -SL --output dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet/Sdk/2.2.103/dotnet-sdk-2.2.103-linux-arm.tar.gz`
+  * 3.0 ARM32: `curl -SL --output dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet/Sdk/3.0.100-preview-010184/dotnet-sdk-3.0.100-preview-010184-linux-arm.tar.gz`
+  * 3.0 ARM64: `curl -SL --output dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet/Sdk/3.0.100-preview-010184/dotnet-sdk-3.0.100-preview-010184-linux-arm64.tar.gz`
 
-For Raspbian [Debian 9 Jessie](https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x#install-net-core-for-debian-8-or-debian-9-64-bit) you need to do the following:
-```
-sudo apt-get update
-sudo apt-get install curl libunwind8 gettext apt-transport-https
+  Note: The latest download links are available at the [.NET Core Download page](https://dotnet.microsoft.com/download/archives).
+2. Unpack .NET Core to a global location:
```
Comment (MichaelSimons) [inline samples/RaspberryPiInstructions.md]: The formatting is not correct here.  When rendered on GitHub, '2.' is not on a separate line.

File: samples/RaspberryPiInstructions.md [hunk 4]
```diff
@@ -1,70 +1,129 @@
...
-* Copy your app, i.e. whole `publish` directory mentioned above, to the Raspberry Pi and execute run `./helloworld` to see `Hello World!` from .NET Core running on your Pi! (make sure you `chmod 755 ./helloworld`)
+## Installing the .NET Core SDK on Raspberry Pi on Win10 IoT Core
+
+Install [Windows 10 IoT Core](https://docs.microsoft.com/en-us/windows/iot-core/getstarted) on your Pi.
+
+Install the [.NET Core 3.0 SDK for Windows ARM32](https://dotnet.microsoft.com/download/dotnet-core/3.0).
+
+## Creating and Running an Application on the Pi
+
+It is easy to run an application on the Pi. The following commands are for Linux, but switching the casing will enable the same commands to work on Windows.
```
Comment (MichaelSimons) [inline samples/RaspberryPiInstructions.md]: > switching the casing will enable the same commands to work on windows

I'm not sure I follow this statement?  What casing are you referring to?

File: samples/RaspberryPiInstructions.md [hunk 5]
```diff
@@ -1,70 +1,129 @@
...
+2. `cd webapp`
+3. `dotnet run` // Build and run app -- will use `localhost:5000`
+4. `\bin\Debug\netcoreapp3.0\webapp.exe` // Run app directly (once built)
+
+Now, we will copy the app to the Pi with scp (other tools work similarly, like Putty). There are a few different options.
+
+Assume matching .NET Core runtime is installed on Pi:
 
-### Win10 IoT Core
+1. `scp -r bin\Debug\netcoreapp3.0\ pi@192.168.1.75:~/webapp` // copy app to Pi
```
Comment (JunTaoLuo) [inline samples/RaspberryPiInstructions.md]: Why are we copying over the build output? What we should be doing here is to do a portable publish (FDD) and copying the published output to the deployment machine.

File: samples/RaspberryPiInstructions.md [hunk 6]
```diff
@@ -1,70 +1,129 @@
...
+Web App:
+
+1. `dotnet new webapp -o webapp` // Generate app from template
+2. `cd webapp`
+3. `dotnet run` // Build and run app -- will use `localhost`
+4. `./bin/Debug/netcoreapp3.0/webapp --urls "http://*:5100"` //Run app, exposed on port 5100
+
+Note: `dotnet run` will use the `localhost` loopback by default. To view the site on an external port, [configure ASP.NET Core](https://docs.microsoft.com/aspnet/core/fundamentals/host/web-host?view=aspnetcore-2.1#server-urls) to do so with `--urls "http://*:5100"` argument (with `dotnet run` or direct exectuable launch, as seen in step 4 above) and then view on the appropriate IP address and port, such as `http://192.168.1.75:5100`.
+
+## Build on X64 and Publish to Pi
```
Comment (MichaelSimons) [inline samples/RaspberryPiInstructions.md]: Thoughts on including instructions for Windows ARM?

File: samples/RaspberryPiInstructions.md (hunk 5 above)
Comment (richlander) [reply samples/RaspberryPiInstructions.md]: In 3.0, you are supposed to be able to copy build output. Does that not work for asp.net?

File: samples/RaspberryPiInstructions.md [hunk 7]
```diff
@@ -1,70 +1,129 @@
...
+```console
+dotnet
 ```
 
-* Copy your app, i.e. whole `publish` directory mentioned above, to the Raspberry Pi and execute run `./helloworld` to see `Hello World!` from .NET Core running on your Pi! (make sure you `chmod 755 ./helloworld`)
+## Installing the .NET Core SDK on Raspberry Pi on Win10 IoT Core
+
+Install [Windows 10 IoT Core](https://docs.microsoft.com/en-us/windows/iot-core/getstarted) on your Pi.
+
+Install the [.NET Core 3.0 SDK for Windows ARM32](https://dotnet.microsoft.com/download/dotnet-core/3.0).
```
[the remark this answers is not in the document]
Comment (richlander) [reply samples/RaspberryPiInstructions.md]: Cuz I've never done it before, even though that is very bad of me. Either I'll merge and do it later when I have time or someone else will be nice and fill this in.

File: samples/RaspberryPiInstructions.md (hunk 4 above)
Comment (richlander) [reply samples/RaspberryPiInstructions.md]: Crap, I meant the slash direction. I'll fix that.

File: samples/RaspberryPiInstructions.md [hunk 8]
```diff
@@ -1,70 +1,129 @@
...
+  * 3.0 ARM32: `curl -SL --output dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet/Sdk/3.0.100-preview-010184/dotnet-sdk-3.0.100-preview-010184-linux-arm.tar.gz`
+  * 3.0 ARM64: `curl -SL --output dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet/Sdk/3.0.100-preview-010184/dotnet-sdk-3.0.100-preview-010184-linux-arm64.tar.gz`
 
-For Raspbian [Debian 9 Jessie](https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x#install-net-core-for-debian-8-or-debian-9-64-bit) you need to do the following:
-```
-sudo apt-get update
-sudo apt-get install curl libunwind8 gettext apt-transport-https
+  Note: The latest download links are available at the [.NET Core Download page](https://dotnet.microsoft.com/download/archives).
+2. Unpack .NET Core to a global location:
+  * `sudo mkdir -p /usr/share/dotnet`
```
[the remark this answers is not in the document]
Comment (richlander) [reply samples/RaspberryPiInstructions.md]: I was copying these commands. Are they wrong? https://github.com/dotnet/dotnet-docker/blob/master/3.0/sdk/stretch/arm64v8/Dockerfile

File: samples/RaspberryPiInstructions.md (hunk 5 above)
Comment (JunTaoLuo) [reply samples/RaspberryPiInstructions.md]: Ah, I see. I was under the impression you'd want this to work in 2.2 as well and I was thinking along those lines.

File: samples/RaspberryPiInstructions.md (hunk 1 above)
Comment (richlander) [reply samples/RaspberryPiInstructions.md]: This pattern only works with 3.0

File: samples/RaspberryPiInstructions.md (hunk 5 above)
Comment (richlander) [reply samples/RaspberryPiInstructions.md]: Good point. It's tough to make the docs work for 2.2 vs 3.0, windows vs linux and 32bit vs 64bit. I will endeavour to make that work.

**Comment by karelz** [comment]
@richlander do you have plans to move the PR forward?

**Comment by karelz** [comment]
Abandoned PR, closing. @richlander feel free to reopen when you're ready to move it forward. Thanks!

**Pull request closed without merging**